### ISSUE TITLE
Google calendar sidebar fails to load

### ISSUE THREAD

**Issue by ShaunGriffithRyff** [polcak/jsrestrictor#119]
Google calendar sidebar fails to load, even with google.com whitelisted (so it's probably not that).

Chrome Version 94.0.4606.61 (Official Build) (x86_64) on Mac OS X 11.6.

**Comment by FedX-sudo** [comment]
I have noticed the similar issues. It appears that the extensions treat subdomains like `docs.google.com` as entirely separate domains when blocking scripts, and does not allow cross site scripts in blocking. So it will say that `google.com` is whitelisted, but if `callendar.google.com` calls a script from `drive.google.com` it will be blocked by jsrestrictor. Does this help?

**Comment by polcak** [comment]
Do you get notifications about blocked requests? If so, this is caused by Network Boundary Shield, see https://www.fit.vutbr.cz/~ipolcak/jsr/doxygen/html/group__NBS.html for docs (the text should be also available on jshelter.org, I am investigating why it is not listed). https://jshelter.org/localportscanning/ shows another example how NBS works.

If you do not see NBS notifications, whitelisting a domain for NBS does nothing.

> I have noticed the similar issues. It appears that the extensions treat subdomains like `docs.google.com` as entirely separate domains when blocking scripts, and does not allow cross site scripts in blocking. So it will say that `google.com` is whitelisted, but if `calendar.google.com` calls a script from `drive.google.com` it will be blocked by jsrestrictor. Does this help?

If you set a specific level for google.com, it should be the same for all subdomains of google.com. What level is applied is based on the visited domain (page URL). It does not matter from what domain are the included scripts. All share the same JS environment.

If a page has several iframes, all iframes have specific environments, so different level might be applied to each iframe, depending on the settings. Again all scripts in an iframe will see the same wrapped objects.

**Comment by martinbednar** [comment]
@ShaunGriffithRyff, @FedX-sudo
I am sorry, we can not reproduce this bug. Could you provide more information?
What JSR level do you use? Can you attach printscreen of the broken page?

**Comment by FedX-sudo** [comment]
![Screenshot_20211008_074350](https://user-images.githubusercontent.com/66258975/136567944-3dc71dfc-8470-4f19-b82b-75c9e7136256.png)

I have the extension set to "off" for all google.com domains, so it does not let me set a JSR level.

**Comment by polcak** [comment]
Is this still valid? It seems to me that you used XMLHttpRequest blocking level by mistake. We no longer activate XHR protections in the Strict level as it was superseded by Fingerprint Detector.

**Comment by polcak** [comment]
Closing for a lack of activity. Use JShelter in Brave with care, see https://jshelter.org/faq/#should-i-install-jshelter-if-i-am-using-brave-browser.
